src/experimental/Toast/ToastUI.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import {CheckIcon, AlertIcon, StopIcon, InfoIcon, XIcon} from '@primer/octicons-react'
+
+import {Spinner, Box, IconButton} from '../..'
+
+const variants = ['default', 'success', 'attention', 'danger', 'loading'] as const
+
+const icon: Record<ToastVariants, React.ReactNode> = {
+  default: <InfoIcon size="small" />,
+  success: <CheckIcon size="small" />,
+  attention: <AlertIcon size="small" />,
+  danger: <StopIcon size="small" />,
+  loading: <Spinner size="small" />,
+}
+
+export const ToastUI = ({
+  dismissable = false,
+  variant = 'default',
+  role = 'status',
+  dismissLabel = 'Dismiss this message',
+  children,
+}: React.PropsWithChildren<ToastUIProps>) => {
+  if (__DEV__) {
+    // eslint-disable-next-line no-console
+    console.assert(
+      variants.includes(variant),
+      `The variant ${variant} is not supported. Use one of ${variants.join(', ')}`,
+    )
+  }
+
+  return (
+    <Box role={role} sx={{display: 'flex', justifyContent: 'flex-start'}}>
+      <Box
+        sx={{display: 'inline-flex', maxWidth: '450px', borderRadius: 2, boxShadow: 'shadow.large', overflow: 'hidden'}}
+      >
+        <Box
+          sx={{
+            display: 'flex',
+            justifyItems: 'center',
+            alignItems: 'center',
+            borderTopLeftRadius: 2,
+            borderBottomLeftRadius: 2,
+            border: '1px solid',
+            borderColor: 'border.default',
+            borderInlineEnd: 'none',
+            p: 3,
+            color: 'white',
+            '&[data-variant="default"]': {
+              backgroundColor: 'accent.fg',
+              borderColor: 'accent.muted',
+            },
+            '&[data-variant="success"]': {
+              backgroundColor: 'success.fg',
+              borderColor: 'success.muted',
+            },
+            '&[data-variant="attention"]': {
+              backgroundColor: 'attention.fg',
+              borderColor: 'attention.muted',
+            },
+            '&[data-variant="danger"]': {
+              backgroundColor: 'danger.fg',
+              borderColor: 'danger.muted',
```

Comment:
```suggestion
            '&[data-variant="default"]': {
              backgroundColor: 'accent.emphasis',
              borderColor: 'accent.emphasis',
            },
            '&[data-variant="success"]': {
              backgroundColor: 'success.emphasis',
              borderColor: 'success.emphasis',
            },
            '&[data-variant="attention"]': {
              backgroundColor: 'attention.emphasis',
              borderColor: 'attention.emphasis',
            },
            '&[data-variant="danger"]': {
              backgroundColor: 'danger.emphasis',
              borderColor: 'danger.emphasis',
```

I'm making this suggestion without testing, but I think we want `emphasis` and we also want the background and border color to match.

Its easier to tell that the background color is incorrect if you check in the high contrast modes. I'm curious to see how the emphasis color looks in dark high contrast, as right now the white doesn't look correct over the lighter yellow shade.

![toast component in dark high contrast mode](https://github.com/primer/react/assets/18661030/af10a9f3-579a-44f8-97a6-039cabe8c4a4)